overlord/snapstate/snapmgr.go

```diff
@@ -389,24 +399,24 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 			panic("internal error: invalid local revision built: " + revision.String())
 		}
 		snapst.LocalRevision = revision
-		ss.Revision = revision
-		snapst.Candidate = &snap.SideInfo{Revision: ss.Revision}
+		ss.SideInfo.Revision = revision
+		snapst.Candidate = &snap.SideInfo{Revision: ss.Revision()}
 	} else {
 		for _, si := range snapst.Sequence {
```

Comment:
wondering if in the new world we should just set for a revert  ss.SideInfo to the full content of Sequence[i] in revertToRevision and here just do:

snapst.Candidate = ss.SideInfo